app/questionnaire/value_source_resolver.py

```diff
@@ -162,25 +184,27 @@ def resolve(
     ) -> Union[ValueSourceEscapedTypes, ValueSourceTypes]:
         source = value_source["source"]
 
-        if source == "answers":
-            return self._resolve_answer_value_source(value_source)
-
-        if source == "list":
-            return self._resolve_list_value_source(value_source)
-
-        if source == "metadata":
+        def resolve_metadata_source(_value_source: Mapping) -> Optional[str]:
```

Comment:
Why is the function arg using underscore?

Reply:
In order to avoid a variable name conflict from the upper scope since it is a nested function.

Reply:
I don't get why 3 of those methods are internal and others aren't, make them all external for consistency?